### ISSUE TITLE
Bug: [no-shadow] Enum members flagged as shadowed vars

### ISSUE THREAD

**Issue by niekvb** [typescript-eslint/typescript-eslint#5985]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.8.4&sourceType=module&code=MYewdgzgLgBAwuaMC8MBEpJTQbgLABQhApmAK4C2MAouVQN6EzyKyoatoA0hAvoUA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQHYHsBaZACwEMATfAd3TAG1J8AzZyAXQBpxtIABAC4BPAA4oAxtACWIgYRTwpuAQHoCxclVoZGiaNHzQOPAL4gTQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
const Const = "const";

enum Enum {
    Const = "const",
//  ^^^^^ 'Const' is already declared in the upper scope on line 1 column 7. eslint@typescript-eslint/no-shadow
}
```

### ESLint Config

```javascript
{
  "rules": {
    "no-shadow": ["off"],
    "@typescript-eslint/no-shadow": ["error"]
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

No errors

### Actual Result

`'Const' is already declared in the upper scope on line 1 column 7. eslint@typescript-eslint/no-shadow`

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.42.1` |
| `@typescript-eslint/parser`            | `5.42.1` |
| `TypeScript`                           | `4.7.4` |
| `ESLint`                               | `8.27.0` |
| `node`                                 | `v18.11.0` |

**Comment by bradzacher** [comment]
> - [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22+sort%3Aupdated-desc) and found none that matched my issue.

[Duplicate of many, many issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+%22no-shadow%22+is%3Aclosed+in%3Atitle+label%3A%22working+as+intended%22+enum+sort%3Aupdated-desc).
Please use the search for opened and closed issues.
